src/api/creatures.py

```diff
@@ -248,69 +248,74 @@ class NewCreature(BaseModel):
     creature_id_2: int
     name: str
 
-@router.post("/{creature_id}/breed")
+@router.post("/breed")
 def breed_creatures(user_id: int, new: NewCreature):
     """ 
     Breeds 2 creatures together. Creatures must be adopted by the user.
     """
 
-    with db.engine.begin() as connection:
-        result = connection.execute(
-            sqlalchemy.text("""SELECT creatures.type, fav_treat, hated_treat
-                                FROM creatures 
-                                JOIN user_creature_connection 
-                                ON creatures.id = creature_id 
-                                JOIN creature_types 
-                                ON creatures.type = creature_types.type
-                                WHERE user_id = :uid 
-                                AND (creatures.id = :cid1
-                                OR creatures.id = :cid2)
-                                AND is_adopted = TRUE"""),
-                                {"uid": user_id, "cid1": new.creature_id_1, "cid2": new.creature_id_2}).mappings().fetchall()
+    try: 
+        with db.engine.begin() as connection:
+            result = connection.execute(
+                sqlalchemy.text("""SELECT creatures.type, fav_treat, hated_treat
+                                    FROM creatures 
+                                    JOIN user_creature_connection 
+                                    ON creatures.id = creature_id 
+                                    JOIN creature_types 
+                                    ON creatures.type = creature_types.type
+                                    WHERE user_id = :uid 
+                                    AND (creatures.id = :cid1
+                                    OR creatures.id = :cid2)
+                                    AND is_adopted = TRUE"""),
+                                    {"uid": user_id, "cid1": new.creature_id_1, "cid2": new.creature_id_2}).mappings().fetchall()
 
-        if not result or len(result) != 2:
-            return {"success": False, "error": f"Cannot breed creatures"}
-        
-        # determine new type     
-        type1 = result[0]["type"].split('_')
-        type2 = result[1]["type"].split('_')
-        new_type = type1[0] + "_" + type2[1]
+            if not result or len(result) != 2:
+                return {"success": False, "error": f"Creature(s) are not adopted or do not exist"}
```

Comment:
ideally, this would be separate errors to make the transition to HTTP statuses easier.